meta-balena-common/recipes-containers/docker-disk/docker-disk.inc

```diff
@@ -15,24 +15,26 @@ def connected(d):
         pass
     return "no"
 
-# Check if docker is running and usable for current user
-def usable_docker(d):
+# Check if the engine is running and usable for current user
+def usable_engine(d):
+    engine_client = d.getVar('ENGINE_CLIENT', True)
     import subprocess
 
-    # Test docker execute permission
-    cmd = "docker images > /dev/null 2>&1"
+    # Test engine execute permission
+    cmd = engine_client + " images > /dev/null 2>&1"
     child = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
     cmd_output = child.communicate()[0]
     if child.returncode != 0:
         return "no"
 
     return "yes"
 
-python () {
-    # We need docker on host
-    if usable_docker(d) != "yes":
-        bb.fatal("docker-disk: Docker needs to run on your host and current user must be able to use it.")
+python do_is_engine_usable() {
+    # We need balena engine on host
+    if usable_engine(d) != "yes":
+        bb.fatal("docker-disk: Balena engine needs to run on your host and current user must be able to use it.")
```

Comment:
Is this an actionable error? When would we expect to see this and how would it be resolved by a contributor?

Reply:
It would be a bug, like a race condition that we need to fix. The rootless engine is started in by the balena-engine-rootless class and launched before do_compile(), so it should be running by the time the build gets here.

Reply:
That's what I thought, so a contributor should never see this and it would be treated as a bug.